Add a README which includes an example of how to add a test (#3384)
singularityware/singularity: e2e/README.md

> For this example, I'm going to use a topic of `env` or `environment variable tests`.

Reviewer: ```suggestion
# End to End Tests

This package contains integration tests. 
The tests work with the `SINGULARITY_E2E` environment variable

## Contributing
For this example, we are going to use a topic of `env` or `environment variable tests`.
```

```
touch e2e/env/env_utils.go
```

* Add a package declaration to your topic's test file that matches what you put in `suite.go`

```
package env
```

* Add a variable to store the testing settings in your topic's test file.

```
import (
        "github.com/kelseyhightower/envconfig"
)

type testingEnv struct {
	// base env for running tests
	CmdPath     string `split_words:"true"`
	TestDir     string `split_words:"true"`
	RunDisabled bool   `default:"false"`
}

var testenv testingEnv
```

* Add a entry-point to your topic's test file that matches what you put in `suite.go`

```
//RunE2ETests is the main func to trigger the test suite
func RunE2ETests(t *testing.T) {
	err := envconfig.Process("E2E", &testenv)
	if err != nil {
		t.Fatal(err.Error())
	}
	t.Log(testenv)
}
```

* Create a test in your topic's test file as you normally would in `go`.

```
func TestEnv(t *Testing.T) {
	...
}
```

* Run your test from your entry-point function using a `go` sub-test.
```
//RunE2ETests is the main func to trigger the test suite
func RunE2ETests(t *testing.T) {
	err := envconfig.Process("E2E", &testenv)
        if err != nil {
        	t.Fatal(err.Error())
        }
        

> 	t.Log(testenv)

Reviewer: ```suggestion

```
This was for development and debugging.  I will rm from the current code now that is running

	// Add tests
	t.Run("TestEnv", TestEnv)
}
```

* Example of what your topic's test file might look like:

```
package env 

import (
	"github.com/kelseyhightower/envconfig"
)

type testingEnv struct {
	// base env for running tests
	CmdPath     string `split_words:"true"`
	TestDir     string `split_words:"true"`
	RunDisabled bool   `default:"false"`
}

var testenv testingEnv

func TestEnv(t *testing.T) {
	...
}

//RunE2ETests is the main func to trigger the test suite
func RunE2ETests(t *testing.T) {
	err := envconfig.Process("E2E", &testenv)
	if err != nil {
		t.Fatal(err.Error())
	}

> 	t.Log(testenv)

Reviewer: ```suggestion
	
```
same as above

	t.Run("TestEnv", TestEnv)
}
```

> * Test your topic using the `e2e` target in the `Makefile`

Reviewer: ```suggestion
## Running the E2E tests

Test your topic using the `e2e` target in the `Makefile`
```
